src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -621,31 +618,50 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
+        InputStream fileInputStream = null;
         try {
-            Path path = IOUtil.getPath(potentialInput);
-            //isVCFStream closes the stream that's passed in
-            return isVCFStream(Files.newInputStream(path), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(Files.newInputStream(path)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(Files.newInputStream(path)), MAGIC_HEADER_LINE);
+            final Path path = IOUtil.getPath(potentialInput);
+            fileInputStream = Files.newInputStream(path);
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream);
+            bufferedInputStream.mark(0);
```

Comment:
Could you only pass in one inputstream into `isVCFStream` and reset here rather than within isVCFStream?

the code here would then look like:
```
bufferedInputStream.mark(0);
if isVCFStream(bufferedInputStream, MAGIC_HEADER_LINE) return true;
bufferedInputStream.reset(0);
if isVCFStream(new GZIPInputStream(bufferedInputStream), MAGIC_HEADER_LINE) return true;
bufferedInputStream.reset(0);
if isVCFStream(new BlockCompressedInputStream(bufferedInputStream), MAGIC_HEADER_LINE) return true;
return false;